includes/admin/upgrades/upgrade-functions.php

```diff
@@ -909,19 +918,125 @@ function give_v188_upgrades() {
 
 	global $wp_roles;
 
-    // Get the role object.
-    $give_worker = get_role( 'give_worker' );
+	// Get the role object.
+	$give_worker = get_role( 'give_worker' );
 
 	// A list of capabilities to add for give workers.
-    $caps_to_add = array(
-        'edit_posts',
-        'edit_pages'
-    );
+	$caps_to_add = array(
+		'edit_posts',
+		'edit_pages',
+	);
+
+	foreach ( $caps_to_add as $cap ) {
+		// Add the capability.
+		$give_worker->add_cap( $cap );
+	}
+
+}
+
+/**
+ * Update Post meta for minimum and maximum amount for multi level donation forms
+ *
+ * This upgrade routine adds post meta for give_forms CPT for multi level donation form.
+ *
+ * @since      1.8.9
+ */
+function give_v189_upgrades_levels_post_meta_callback() {
+
+	if ( ! current_user_can( 'manage_give_settings' ) ) {
+		wp_die( esc_html__( 'You do not have permission to do Give upgrades.', 'give' ), esc_html__( 'Error', 'give' ), array(
+			'response' => 403,
+		) );
+	}
+
+	ignore_user_abort( true );
+
+	if ( ! give_is_func_disabled( 'set_time_limit' ) && ! ini_get( 'safe_mode' ) ) {
+		@set_time_limit( 0 );
+	}
+
+	$step = isset( $_GET['step'] ) ? absint( $_GET['step'] ) : 1;
+
+	// form query
+	$donation_forms = new WP_Query( array(
+			'paged'          => $step,
+			'status'         => 'any',
+			'order'          => 'ASC',
+			'post_type'      => 'give_forms',
+			'posts_per_page' => 20,
+		)
+	);
+
+	if ( $donation_forms->have_posts() ) {
+		while ( $donation_forms->have_posts() ) {
+			$donation_forms->the_post();
+			$form_id = get_the_ID();
 
-    foreach ( $caps_to_add as $cap ) {
-        // Add the capability.
-        $give_worker->add_cap( $cap );
-    }
+			// Remove formatting from _give_set_price
+			update_post_meta(
```

Comment:
@DevinWalker I added upgrade script to sanitize amount values:
1. `_give_set_value`
2. `_give_custom_amount_minimum`
3. `_give_donation_levels[$index][_give_amount]`